### ISSUE TITLE
Parsing error when using reserved keyword

### ISSUE THREAD

**Issue by timvink** [sqlfluff/sqlfluff#1867]
First of all, thanks for all the effort put into this plugin. Really excited about having a `black`-like linter for `sql` ! I also know it's very hard to catch all edge cases and it can help to report any issues you encounter as a user, so hope this issue helps :)

## Expected Behaviour

I would expect no error, or something about using a reserved keyword when running `sqlfluff lint query.sql`: 

```
# .sqlfluff
[sqlfluff]
dialect = tsql
```

```
-- query.sql
SELECT sometable.column
FROM sometable
WHERE 1 = 1

```

## Observed Behaviour

Instead, I get:

```
matching:   0%|                                                                       | 0/2 [00:00<?, ?it/s]== [query.sql] FAIL
L:   1 | P:  18 |  PRS | Line 1, Position 18: Found unparsable section:
                       | 'column\nFROM sometable\nWHERE 1 = 1\n'
All Finished �📜🎉🎉!
```

(yes, I'm using powershell on my corporate laptop and it has some encoding issues, but it's not a problem. The 🎉 's are maybe not appriorate on a fail ;) )

Interestingly, when I changed the columnname from `sometable.column` to `sometable.some_column` there is no parsing error:

```
-- query.sql
SELECT sometable.some_column
FROM sometable
WHERE 1 = 1

```

```
matching:   0%|                                                                       | 0/2 [00:00<?, ?it/s]
All Finished �📜🎉🎉!                                                                                       ]
```

## Steps to Reproduce

```shell
mkdir test_sql
cd test_sql
echo "[sqlfluff]\ndialect = tsql" > .sqlfluff
echo "SELECT sometable.column\nFROM sometable\n" > query.sql
pip install sqlfluff==0.8.1
sqlfluff lint query.sql
```

## Dialect

`tsql`

## Version

`Python 3.9.5`
`sqlfluff, version 0.8.1`

## Configuration

```
# .sqlfluff
[sqlfluff]
dialect = tsql
```

**Comment by WittierDinosaur** [comment]
So, the thing with this kind of issue, is that that code wouldn't run on your database either. You have to quote reserved keywords to use them. There's an argument to be made that this shouldn't be a parsing error - but technically the SQL language can't parse it either 🤷‍♂️

**Comment by timvink** [comment]
That's fair enough. The list of reserved keywords is quite long and users can make mistakes (I did) -- it helps if the linter tells you what's wrong with your query, not only that it's broken. I thought it was a problem with Sqlfluff.

Closing the issue -- thanks for the explanation!
